src/main/java/org/broadinstitute/hellbender/tools/spark/ApplyBQSRSpark.java

```diff
@@ -48,15 +49,22 @@
     @Argument(doc = "If specified, shard the output bam", shortName = "shardedOutput", fullName = "shardedOutput", optional = true)
     private boolean shardedOutput = false;
 
+    @Argument(doc="The output parallelism, sets the number of reducers. Defaults to one partition per 10MB of input.",
+            shortName = "P", fullName = "parallelism", optional = true)
+    protected int parallelism = 0;
```

Comment:
I think this argument should be moved up into `GATKSparkTool` to cut down on code duplication. It's ok if it's not applicable to all Spark tools -- we can make this clear in the docs. Eg., "For tools that write an output, sets the number of reducers...".

Also, we might want to rename it to be a little more self-explanatory. Perhaps `numOutputPartitions`?